File: 01-no-tools.py
import json
import boto3
import asyncio

MODEL_ID = "us.amazon.nova-lite-v1:0"
INITIAL_PROMPT = "Navigate to AWS homepage and take a screenshot. Do the same for Anthropic homepage"

async def run_example():
    # Set up AWS Bedrock client
    bedrock_client = boto3.client('bedrock-runtime')
    
    messages = [{
        "role": "user",
        "content": [{"text": INITIAL_PROMPT}]
    }]
    nb_request = 1
    # Send to model
    print(f"Sending request {nb_request} to Bedrock with {len(messages)} messages...")
    print(f"User prompt: {messages[0]['content'][0]['text']}")
    response = bedrock_client.converse(
        modelId=MODEL_ID,
        messages=messages,
    )
    
    # Process response
    output_message = response.get('output', {}).get('message', {})
    print(f"Model response {json.dumps(output_message, indent=2)}")


# Main entry point
if __name__ == "__main__":
    print("AWS Bedrock Web Tools Minimal Example")
    print("------------------------------------")
    asyncio.run(run_example())

File: JavaScript/01-no-tools.js
import { BedrockRuntimeClient, ConverseCommand } from "@aws-sdk/client-bedrock-runtime";
import chalk from 'chalk';

const modelId = "us.amazon.nova-lite-v1:0";
const initialPrompt = "Navigate to AWS homepage and take a screenshot. Do the same for Anthropic homepage";

function printUser(s) {
    console.log(chalk.blue(s));
}
function printAssistant(s) {
    console.log(chalk.red(JSON.stringify(s, null, 2)));
}
function printSystem(s) {
    console.log(chalk.green(JSON.stringify(s, null, 2)));
}
async function run() {
    // Set up AWS Bedrock client
    const bedrockClient = new BedrockRuntimeClient({ region: "us-west-2" });

    try {
        const messages = [
            {
                role: "user",
                content: [
                    {
                        text: initialPrompt
                    }
                ]
            }
        ];

        let nbRequest = 1;
        // Send to model
        printSystem(`Sending request ${nbRequest} to Bedrock with ${messages.length} messages...`)
        printUser(`User prompt: ${messages[0]['content'][0]['text']}`)

        let response = await bedrockClient.send(
            new ConverseCommand({
                modelId,
                messages: messages
            })
        );

        // Process response
        printAssistant(response.output.message);
    } catch (error) {
        console.error("Error invoking Bedrock model:", error);
    }
}

// Main entry point
printSystem("AWS Bedrock Web Tools Minimal Example")
printSystem("------------------------------------")
run();
